### ISSUE TITLE
Closing all files in editor shouldn't left it empty

### ISSUE THREAD

**Issue by Garrett-R** [spyder-ide/spyder#2302]
Open a file in the editor, then click the small Options icon in the top-right and click `Split Horizontally`.  The file should appear on both sides.  Now close the file and this will have the effect of closing the extra panel.  That is, if you were to open a new file, you now only have 1 panel.

I think this is unexpected behavior and not in line with other programs such as my C++ IDE, Qt Creator.

I think both the panel layout and the small Options button should persist even when no files are open in the editor.  As it is currently, if I'm closing files, because I'm about to work on a different project, I have to make sure to open the new project files before closing all the current ones.

(I'm on Ubuntu 14.04.2 and the latest version of Spyder, i.e. 91dac51)

**Comment by ccordoba12** [comment]
This is not easy to fix. If there are no files open, I don't know how we could maintain the splitted view.

The only solution I can think of is to forbade closing the last opened file.

**Comment by Garrett-R** [comment]
I see.  I think preventing the last open file from closing would be bad...

**Comment by ccordoba12** [comment]
That would work like in Firefox, i.e. just the last file couldn't be closed.

Another option would be to open a new file when closing the last one.

**Comment by Garrett-R** [comment]
My opinion is that both options would be more surprising than the original behavior of closing the panels (which isn't that a huge problem).  I've never seen any IDE that automatically open a new file when the last one got closed.

And Firefox (and Chromium) actually terminates when the last tab is closed, probably because it's a clear indication that the user has finished with the application, but in Spyder, I don't think that's the case.  The user may still have a terminal session open, for example.

**Comment by Shaunakde** [comment]
@ccordoba12 and @Garrett-R  Perhaps opening a new blank tab is the best solution. It is the default behavior of Notepad++ and is not jarring at all.

**Comment by ccordoba12** [comment]
Yes, I think that too!

@blink1073, could you take a look at this one when you have some time?

**Comment by Shaunakde** [comment]
@ccordoba12 I'd like to try and look at this too. Could you tell where to roughly start looking?

**Comment by blink1073** [comment]
It seems like *something* would have to take the closed dock's place, perhaps an empty dock window?

**Comment by ccordoba12** [comment]
I think this is simpler:

1. Check how many files are open (on the current Editor Stack) in the `close_file` method of `plugins/editor`
2. If there is only one file open and the user wants to close it, call the `new` method after that to create a new file.

I think that's all :-)

**Comment by blink1073** [comment]
Yeah I'm with you @ccordoba12, this would be akin to emacs opening a new buffer in a split view.

**Comment by ccordoba12** [comment]
@blink1073, oh yeah!! Besides opening a new file, we should split it as before :-)

**Comment by goanpeca** [comment]
@Shaunakde if you still want to help on this, you can start taking a look at [this](https://github.com/spyder-ide/spyder/blob/master/spyderlib/plugins/editor.py#L1713) and try  to implement @ccordoba12 suggestions.

**Comment by Shaunakde** [comment]
I will take a look as soon as I get home tonight. Thank you!

On Wed, Apr 22, 2015 at 6:29 PM, Gonzalo Peña-Castellanos <
notifications@github.com> wrote:

> @Shaunakde <https://github.com/Shaunakde> if you still want to help on
> this, you can start taking a look at this
> <https://github.com/spyder-ide/spyder/blob/master/spyderlib/plugins/editor.py#L1713>
> and try to implement @ccordoba12 <https://github.com/ccordoba12>
> suggestions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spyder-ide/spyder/issues/2302#issuecomment-95256025>.
>



-- 
Shaunak De
http://shaunak.de/

**Comment by goanpeca** [comment]
:+1: great, remember to drop by [gitter](https://gitter.im/spyder-ide) if you have doubts, cheers

**Comment by ccordoba12** [comment]
I invested a couple of hours trying to preserve panel distribution after closing all files, but I couldn't make it work reliably. So I'm sorry to say we're not going to be implement that feature.

I just implemented the simplest thing, which is opening a new empty file after closing all files. This way the Editor will never be empty again.

Since projects are going to revamped in 3.0, and will let users to easily switch between sessions (somethings we're not offering right now), the feature required by @Garrett-R won't be really necessary :-) I mean, panel distribution will be saved for every project.
